### PR TITLE
Give `TreehouseView` a generic for its Widget type

### PR DESCRIPTION
**Pull request by veyndan**
A PR to showcase what https://github.com/cashapp/redwood/issues/883#issuecomment-1711507473 could look like.

### CODE REVIEW

File: redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseView.kt [hunk 1]
```diff
@@ -24,22 +24,22 @@ import kotlinx.coroutines.flow.StateFlow
 import kotlinx.serialization.json.Json
 
 @ObjCName("TreehouseView", exact = true)
-public interface TreehouseView {
-  public val children: Widget.Children<*>
+public interface TreehouseView<W : Any> {
+  public val children: Widget.Children<W>
```
Comment (veyndan) [inline redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseView.kt:28]: Having `children` parametrised is technically the only requirement for having `TreehouseView` extend from `RedwoodView`. The other parametrisations within the file are purely in the hope of improving type safety.

**Review by JakeWharton — APPROVED** [review]
I think this is an improvement. Long term I'd still like to investigate eliminating TreehouseView subtypes in favor of a single wrapper now that all the complexity was pushed to RedwoodView.

**Comment by veyndan** [comment]
How would we eliminate `TreehouseView` subtypes, when the implementation of `TreehouseView`-specific fields/methods are platform-specific (e.g., in `TreehouseLayout` we have to override `onAttachedToWindow` in order to set `readyForContent = true`).

**Comment by JakeWharton** [comment]
Expose the attached stats and also use it to pause the recomposition for plain Redwood.

**Pull request merged**